### PR TITLE
RavenDB-12967 Race condition in data exploration means that we aren't…

### PR DESCRIPTION
**Pull request by ayende**
… sending any results out

- Underlying issue was that we were disposing the cancellation token too early

### CODE REVIEW

**Comment by ayende** [comment]
The test failure isn't related to the changes

File: Raven.Database/Server/Controllers/StreamsController.cs [hunk 1]
```diff
@@ -246,69 +246,67 @@ public async Task<HttpResponseMessage> Exploration(string collection)
 
             var msg = GetEmptyMessage();
 
-            using (var cts = new CancellationTokenSource())
+            var cts = new CancellationTokenSource();
```
Comment (talweiss1982) [inline Raven.Database/Server/Controllers/StreamsController.cs:249]: Shouldn't we dispose of the Cts? maybe we need to override the Dispose of the HttpContent and dispose of the Cts there.

File: Raven.Database/Server/Controllers/StreamsController.cs [hunk 2]
```diff
@@ -246,69 +246,67 @@ public async Task<HttpResponseMessage> Exploration(string collection)
...
+            var timeout = cts.TimeoutAfter(TimeSpan.FromSeconds(timeoutSeconds));
```
Comment (talweiss1982) [inline Raven.Database/Server/Controllers/StreamsController.cs:250]: We are also seems to be leaking a timer here

File: Raven.Database/Server/Controllers/StreamsController.cs (hunk 1 above)
Comment (ayende) [reply Raven.Database/Server/Controllers/StreamsController.cs:249]: No, `StreamQueryContent` will handle that for us.

File: Raven.Database/Server/Controllers/StreamsController.cs (hunk 2 above)
Comment (ayende) [reply Raven.Database/Server/Controllers/StreamsController.cs:250]: This is handled by `StreamQueryContent`, if there is an error here that isn't caught, we'll let the finalizer handle this.
Expected to be very rare

**Pull request merged**